### ISSUE TITLE
False positives on SC3028

### ISSUE THREAD

**Issue by aep-sunlife** [koalaman/shellcheck#3156]
Hi,

I am using POSIX safe code in my shell scripts, but ShellCheck insists on presenting some spurious warnings.

```sh
if [ "${HOSTNAME:-}" = 'beavis' ]; then
    export JAVA_HOME="/opt/jenkins/java/jdk-${JAVA_VERSION}"
fi
```

```text
In scripts/provision-jenkins.sh line 1:
if [ "${HOSTNAME:-}" = 'beavis' ]; then
      ^-----------^ SC3028 (warning): In POSIX sh, HOSTNAME is undefined.
```

ShellCheck should recognize that we're not hard depending on the existence of the `HOSTNAME` environment variable, but in fact falling back on POSIX's default value syntax. So no warning should emit.
